types/index.d.ts

```diff
@@ -278,7 +278,7 @@ declare module 'mongoose' {
      * Returns a list of indexes that this schema declares, via `schema.index()`
      * or by `index: true` in a path's options.
      */
-    indexes(): Array<IndexDefinition>;
+    indexes(): Array<[IndexDefinition, IndexOptions]>;
```

Comment:
any specific reason this is `Array<[v, v]>` instead of `[v,v][]`?

Reply:
maybe Array<[v, v]> is more readable?

Reply:
Yeah I think `Array<[v1, v2]>` is slightly more readable, but it's a minor point. We can change this definition if you feel strongly about one syntax or the other. But I'll merge for now because we can always change the definition later.

Reply:
basically asked because the other definition was used for the tests [reference](https://github.com/Automattic/mongoose/pull/13718/files#diff-8603b830eee3be0bc269312df23ca0e252cc91e4cdda393d2ae7a332cbea30d6R1178)

Reply:
Oh yeah I used the other definition in tests as a sanity check to make sure those definitions were equivalent. I would imagine so, but with TS you never know :)